File: banana/frontend/src/api/payment.js
/**
 * 支付相关 API
 */

import client from './client'
import { API_BASE_URL } from '../config/api'

const paymentAPI = {
  /**
   * 创建支付订单
   * @param {string} plan - 套餐名称
   * @param {string|number} price - 价格
   * @returns {Promise<Object>} 订单信息
   */
  createPayment: async (plan, price) => {
    const response = await client.post('/api/payment/create', {
      plan,
      price: parseFloat(price),
    })
    return response.data
  },

  /**
   * 查询支付订单
   * @param {string} orderId - 订单ID
   * @returns {Promise<Object>} 订单信息
   */
  queryPayment: async (orderId) => {
    const response = await client.get(`/api/payment/query/${orderId}`)
    return response.data
  },

  /**
   * 提交支付订单号
   * @param {string} plan - 套餐名称
   * @param {string|number} price - 价格
   * @param {string} account - 用户账号
   * @param {string} orderNumber - 订单号
   * @returns {Promise<Object>} 提交结果
   */
  submitOrder: async (plan, price, account, orderNumber) => {
    const response = await fetch(`${API_BASE_URL}/api/payment/submit-order`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        plan,
        price: parseFloat(price),
        account,
        orderNumber: orderNumber.trim(),
      }),
    })
    const result = await response.json()
    return result
  },
}

export default paymentAPI

File: studio/backend/routes/payment.py
#!/usr/bin/env python3
"""
支付 API 路由
"""

import logging
import traceback
from datetime import datetime
import os
from fastapi import APIRouter, HTTPException
from pydantic import BaseModel
from log_utils import log_info, log_error, log_success, log_api, log_warning

logger = logging.getLogger("支付API")

router = APIRouter(prefix="/api/payment", tags=["支付"])


# ==================== 请求模型 ====================

class CreatePaymentRequest(BaseModel):
    plan: str
    price: float

class SubmitOrderRequest(BaseModel):
    plan: str
    price: float
    account: str
    orderNumber: str


# ==================== API 端点 ====================

@router.post("/create")
async def create_payment(request: CreatePaymentRequest):
    """
    创建支付订单
    
    参数:
    - plan: 套餐名称
    - price: 价格
    """
    try:
        plan = request.plan
        price = request.price
        
        # 生成订单ID
        order_id = f"ORDER_{datetime.now().strftime('%Y%m%d%H%M%S')}_{os.urandom(4).hex()}"
        
        log_success("支付", "创建支付订单", {"套餐": plan, "价格": price, "订单ID": order_id})
        
        # TODO: 集成实际的支付接口（支付宝等）
        return {
            "success": True,
            "order_id": order_id,
            "plan": plan,
            "price": price,
            "status": "pending",
            "payment_url": None  # 实际支付URL
        }
        
    except Exception as e:
        log_error("支付", "创建支付订单失败", {"错误": str(e)})
        raise HTTPException(status_code=500, detail=str(e))


@router.get("/query/{order_id}")
async def query_payment(order_id: str):
    """
    查询支付订单状态
    
    参数:
    - order_id: 订单ID
    """
    try:
        log_info("支付", "查询支付订单", {"订单ID": order_id})
        
        # TODO: 查询实际的支付订单状态
        return {
            "success": True,
            "order_id": order_id,
            "status": "pending",
            "paid": False
        }
        
    except Exception as e:
        log_error("支付", "查询支付订单失败", {"错误": str(e)})
        raise HTTPException(status_code=500, detail=str(e))


@router.post("/submit-order")
async def submit_order(request: SubmitOrderRequest):
    """
    提交支付订单号
    
    参数:
    - plan: 套餐名称
    - price: 价格
    - account: 用户账号
    - orderNumber: 订单号
    """
    try:
        plan = request.plan
        price = request.price
        account = request.account
        order_number = request.orderNumber
        
        if not order_number:
            raise HTTPException(status_code=400, detail="订单号不能为空")
        
        if not account:
            raise HTTPException(status_code=400, detail="账号不能为空")
        
        # 等级映射
        level_map = {
            "基础版": "basic",
            "专业版": "professional",
            "企业版": "enterprise"
        }
        
        user_level = level_map.get(plan, "normal")
        
        log_success("支付", "收到订单提交", {
            "套餐": plan, "价格": price, "账号": account, "订单号": order_number, "等级": user_level
        })
        
        # TODO: 验证订单号，更新用户等级等
        # 这里可以根据订单号查询支付状态，如果支付成功，则更新用户等级
        
        # 返回成功（实际应用中应该验证订单号并更新用户数据）
        return {
            "success": True,
            "message": "订单提交成功",
            "plan": plan,
            "level": user_level,
            "orderNumber": order_number
        }
        
    except HTTPException:
        raise
    except Exception as e:
        log_error("支付", "提交订单失败", {"错误": str(e)})
        raise HTTPException(status_code=500, detail=str(e))
